nats.go

```diff
@@ -1605,7 +1605,7 @@ func (ac *asyncCallbacksHandler) pushOrClose(f func(), close bool) {
 		ac.head = cb
 	}
 	ac.tail = cb
-	ac.cond.Signal()
+	ac.cond.Broadcast()
```

Comment:
I meant only on close condition. Signal as default is fine.

Reply:
Done.. but to be honest, this is some code that one will scratch their head in few months/years and ask themselves: why is that? ;-)
If there is a chance that we have multiple go routines calling Wait() then it should always be Broadcast(), but that would defeat the purpose of this code anyway since it is meant to ensure that callbacks are dispatched in order.